src/particles/diagnostics/ReducedBeamCharacteristics.cpp

```diff
@@ -136,23 +108,25 @@ namespace impactx::diagnostics
         amrex::ParticleReal const py_mean = values_per_rank_1st.at(5) /= w_sum;
         amrex::ParticleReal const pt_mean = values_per_rank_1st.at(6) /= w_sum;
 
-        std::vector<amrex::ParticleReal> values_per_rank_min = {
-                amrex::get<7>(r), // x_min
-                amrex::get<9>(r), // y_min
-                amrex::get<11>(r), // t_min
-                amrex::get<13>(r), // px_min
-                amrex::get<15>(r), // py_min
-                amrex::get<17>(r), // pt_min
-        };
-
-        std::vector<amrex::ParticleReal> values_per_rank_max = {
-                amrex::get<8>(r), // x_max
-                amrex::get<10>(r), // y_max
-                amrex::get<12>(r), // t_max
-                amrex::get<14>(r), // px_max
-                amrex::get<16>(r), // py_max
-                amrex::get<18>(r), // pt_max
-        };
+        std::vector<amrex::ParticleReal> values_per_rank_min(num_red_ops_1_min);
+
+        /* contains in this order:
+         * x_min, y_min, t_min
+         * px_min, py_min, pt_min
+         */
+        amrex::constexpr_for<0, num_red_ops_1_min> ([&](auto i) {
+            values_per_rank_min[i] = amrex::get<i + num_red_ops_1_sum>(r1);
```

Comment:
I see this in HiPACE++:
```suggestion
            values_per_rank_min[i.value] = amrex::get<i.value + num_red_ops_1_sum>(r1);
```

Reply:
I didn’t know that `std::integral_constant` had a conversion operator to the integer. Without `.value` should work as well. Not sure if this is related to the failing MSVC CI test or not.

Reply:
Copying error for records:
```
     8>D:\a\impactx\impactx\src\particles\diagnostics\ReducedBeamCharacteristics.cpp(118,45): error C2672: 'amrex::get': no matching overloaded function found [D:\a\impactx\impactx\build\lib.vcxproj]
         D:\a\impactx\impactx\build\_deps\fetchedamrex-src\Src\Base\AMReX_Tuple.H(196,1):
         could be 'GpuTupleElement<I,amrex::GpuTuple<Ts...>>::type &&amrex::get(amrex::GpuTuple<Ts...> &&) noexcept'
         	D:\a\impactx\impactx\src\particles\diagnostics\ReducedBeamCharacteristics.cpp(118,51):
         	expression did not evaluate to a constant
         		D:\a\impactx\impactx\src\particles\diagnostics\ReducedBeamCharacteristics.cpp(118,53):
         		failure was caused by a read of a variable outside its lifetime
         		D:\a\impactx\impactx\src\particles\diagnostics\ReducedBeamCharacteristics.cpp(118,53):
         		see usage of 'this'
```

Reply:
@n01r:
```suggestion
            constexpr int idx = i.value + num_red_ops_1_sum;
            values_per_rank_min[i] = amrex::get<idx>(r1);
```